src/util/index.ts

```diff
@@ -35,6 +35,18 @@ export const enumFromValue = <TValue, TEnum extends TValue>(
   throw new Error(`${message} ${value}`)
 }
 
+export const enumKeyFromValue = <TValue, TEnum extends TValue>(
```

Comment:
Can be replaced with `EnumType[enumValue]` in which case it's probably not worth keeping this

Reply:
`BytesBinaryOperator[this.op]` throws `Element implicitly has an 'any' type because expression of type 'BytesBinaryOperator' can't be used to index type 'typeof BytesBinaryOperator' error.`